src/nunit.analyzers.tests/TestCaseSourceUsage/TestCaseSourceUsesStringAnalyzerTests.cs

```diff
@@ -136,5 +136,284 @@ public void Test()
+        }
+
+        [Test]
+        public void AnalyzeWhenNumberOfParametersDoesNotMatch()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    [TestFixture]
+    public class AnalyzeWhenNumberOfParametersDoesNotMatch
+    {
+        [TestCaseSource(↓nameof(TestData), new object[] { 1, 2, 3 })]
+        public void ShortName(int number)
+        {
+            Assert.That(number, Is.GreaterThanOrEqualTo(0));
+        }
+
+        static IEnumerable<int> TestData(string dummy, int anotherDummy)
+        {
+            yield return 1;
+            yield return 2;
+            yield return 3;
+        }
+    }", additionalUsings: "using System.Collections.Generic;");
+
+            var expectedDiagnostic = ExpectedDiagnostic
+                .Create(AnalyzerIdentifiers.TestCaseSourceMismatchInNumberOfParameters)
+                .WithMessage("The TestCaseSource provides '3' parameter(s), but the target method expects '2' parameter(s).");
+            AnalyzerAssert.Diagnostics<TestCaseSourceUsesStringAnalyzer>(expectedDiagnostic, testCode);
+        }
+
+        [Test]
+        public void AnalyzeWhenNumberOfParametersDoesNotMatchInAnotherClass()
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    [TestFixture]
+    public class AnalyzeWhenNumberOfParametersDoesNotMatchInAnotherClass
+    {
+        [TestCaseSource(typeof(AnotherClass), ↓nameof(AnotherClass.TestData), new object[] { 1, 2, 3 })]
+        public void ShortName(int number)
+        {
+            Assert.That(number, Is.GreaterThanOrEqualTo(0));
+        }
+    }
+
+    public class AnotherClass
+    {
+        public static IEnumerable<int> TestData(string dummy, int anotherDummy)
+        {
+            yield return 1;
+            yield return 2;
+            yield return 3;
+        }
+    }", additionalUsings: "using System.Collections.Generic;");
+
+            var expectedDiagnostic = ExpectedDiagnostic
+                .Create(AnalyzerIdentifiers.TestCaseSourceMismatchInNumberOfParameters)
+                .WithMessage("The TestCaseSource provides '3' parameter(s), but the target method expects '2' parameter(s).");
+            AnalyzerAssert.Diagnostics<TestCaseSourceUsesStringAnalyzer>(expectedDiagnostic, testCode);
+        }
+
+        [TestCase("private static readonly TestCaseData[] TestCases = new TestCaseData[0];")]
+        [TestCase("private static TestCaseData[] TestCases => new TestCaseData[0];")]
+        [TestCase("private static TestCaseData[] TestCases() => new TestCaseData[0];")]
+        public void AnalyzeWhenSourceDoesProvideIEnumerable(string testCaseMember)
+        {
+            var testCode = TestUtility.WrapClassInNamespaceAndAddUsing(@"
+    public class AnalyzeWhenSourceDoesProvideIEnumerable
+    {
+        private static readonly TestCaseData[] TestCases = new TestCaseData[0];
```

Comment:
(suggestion, feel free to ignore)
imho, would be a bit more clear if you'd use something like `{testCaseSourceMember}` to make clear that this is a placeholder.
I was a bit confused for a moment why do you replace valid code.
(well, and because I didn't know what `AssertReplace` method does 😄 )

Reply:
I think the suggestion makes sense, so I've corrected all occurrences of this. Thanks 👍